pandas/core/frame.py

```diff
@@ -3995,20 +3995,37 @@ def rename(self, *args, **kwargs):
         We *highly* recommend using keyword arguments to clarify your
         intent.
 
+        Rename columns using a mapping:
         >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
-        >>> df.rename(index=str, columns={"A": "a", "B": "c"})
+        >>> df.rename(columns={"A": "a", "B": "c"})
            a  c
         0  1  4
         1  2  5
         2  3  6
 
-        >>> df.rename(index=str, columns={"A": "a", "C": "c"})
+        Rename rows using a mapping:
```

Comment:
I don't think this example is entirely necessarily, to demonstrate remapping twice.

Reply:
This example is intended to demonstrate renaming the index/rows instead of the columns.

Reply:
I'd be okay with keeping this example if we remap the integer index label to string labels. So `df.rename(index={0: "x",  1: "y", 2: "z"})`

Reply:
ok, nice twist

Reply:
The example assumes that you don't have to create the new dataframe below. In essence, all the examples can just use the first dataframe at the top.

Reply:
New commit pushed.

@mroeschke, can you pay special attention to the correctness of the example with the errors parameter? Don't have the master branch version installed here.